### PR TITLE
added recover and record state functions

### PR DESCRIPTION
**Pull request by lhitch**
Added record_state function and recover_state functions to handle transfer of a plugin state to the database

### CODE REVIEW

File: schema/brain/controller/plugins.py [hunk 1]
```diff
@@ -215,6 +215,18 @@ def quick_change_desired_state(plugin_id, desired_state, conn=None):
         .run(conn)
 
 
+@wrap_rethink_errors
+@wrap_connection
+def recover_state(serv_name, conn=None):
+    curs = RPC.filter({"ServiceName":serv_name}).run(conn)
+    doc = curs.next()
```
Comment (bauman) [inline schema/brain/controller/plugins.py]: may throw a ValueError wrapping ReQL error if the cursor is empty.  May leave a hanging connection if there happens to be more than one document with the same serv_name.   Probably ok, but may consider interating the cursor to completion.

File: schema/brain/controller/plugins.py [hunk 2]
```diff
@@ -215,6 +215,18 @@ def quick_change_desired_state(plugin_id, desired_state, conn=None):
         .run(conn)
 
 
+@wrap_rethink_errors
+@wrap_connection
+def recover_state(serv_name, conn=None):
+    curs = RPC.filter({"ServiceName":serv_name}).run(conn)
```
Comment (bauman) [inline schema/brain/controller/plugins.py]: see https://github.com/ramrod-project/database-brain/pull/117

File: schema/brain/controller/plugins.py [hunk 3]
```diff
@@ -215,6 +215,18 @@ def quick_change_desired_state(plugin_id, desired_state, conn=None):
...
+    doc = curs.next()
+    return doc["PluginState"]
+
+@wrap_rethink_errors
+@wrap_connection
+def record_state(serv_name, state, conn=None):
+    return RPC.filter({"ServiceName":serv_name}).update({"PluginState":state}).run(conn)
```
Comment (bauman) [inline schema/brain/controller/plugins.py]: consider https://github.com/ramrod-project/database-brain/pull/117

**Review by bauman — COMMENTED** [review]
consider these changes

**Review by manziman — APPROVED** [review]

**Pull request merged**